src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -42,26 +42,68 @@ object Homework :
 
     val int = 42
 
-    def not(b: Boolean): Boolean = ??? // here is my greatest solution
+    def not(b: Boolean): Boolean ={
+      if (b) false else true
+    }
+
+    // here is my greatest solution
+
+    def and(left: Boolean, right: Boolean): Boolean =  {
```

Comment:
This method must be "eager", means if `right` is not used then it should not be evaluated.
Hint: the method should correctly work for `and(false, loop)` since the result is obviously `false` and we don't need to calculate `loop`.

Looks like you forgot to commit the tests. So pleas fix existing tests and and add some extra tests explained below.

Please add extra test case to prove "eagerness".
Hint: throw an exception.

Since for "eager" case you know all the arguments, you don't need to use `forAll`, use 
```
property("eager and") = propBoolean { 
   <some predicate> 
}
```
instead.

Reply:
One more hint: try to call this method in the following context:

```
def loop: Boolean = loop

and(false, loop) assert false
```

If `and` is "eager", then `and(false, loop) assert false` completes immediately, if not, it will run forever.

Hint: use call-by-name strategy.

The same for `or`.